src/00/math.asm

```diff
@@ -128,41 +128,49 @@ mul16By8:
 ;; Outputs:
 ;;  DEHL: Product of DE and BC.
 mul16By16:
-    ld hl, 0
-
-    sla e
-    rl d
-    jr nc, $ + 4
-    ld h, b
-    ld l, c
-
-.macro mul16By16Iter
-    add hl, hl
-    rl e
-    rl d
-    jr nc, $ + 6
-    add hl, bc
-    jr nc, $ + 3
-    inc de
-.endmacro
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-    mul16By16Iter
-.undefine mul16By16Iter
+    push bc
```

Comment:
Looks like it destroys HL too, sorry I missed that

Reply:
Output is DEHL, though

Reply:
derp

Reply:
Oh, but I do see that my update still destroys F, let me think this one through